TransformationSystem/Agent/TransformationCleaningAgent.py

```diff
@@ -124,6 +125,49 @@ def initialize(self):
     # # file catalog client
     self.metadataClient = FileCatalogClient()
 
+    # Only at (re)start: will clean ancient transformations (remnants)
+    # 1) get the transformation IDs of jobs that are older than 1 year
+    # 2) find the status of those transformations. Those "Cleaned" and "Archived" will be
+    # cleaned and archived (again)
+    res = JobMonitoringClient().getJobGroups(None, datetime.utcnow() - timedelta(days=365))
```

Comment:
out of curiosity, why redoing cleaned and archived ? 
Isn't that a bit "violent" ? Should it be a script to be launched every now and then ?

Reply:
Basically, it's a race:
- the production manager submits a transformation
- TransformationAgent, and a bit later the WorkflowTaskAgent, put such transformation in their internal queue, so eventually during their (long-ish) cycle they'll work on it.
- 1 minute after creating the transformation, the production manager cleans it (by hand, for whatever reason). So, the status is changed to "Cleaning"
- the TransformationCleaningAgent cleans what has been created (maybe, nothing), sets the transformation status to "Cleaned" or "Archived"
- a bit later the TransformationAgent, and a bit later the WorkflowTaskAgent, kick in, creating tasks and jobs for a production that's effectively cleaned, but these 2 agents don't know yet. 

Of course, one could make one final check in TransformationAgent or WorkflowTaskAgent but these 2 agents are already doing a lot of stuff, and are pretty heavy. So, we should just clean from time to time. What I added here is done only when the agent (re)start, and it's quite light-ish operation anyway.

Reply:
Oh I see. Wouldn't it be easier in step 5. to have these agents checking that the Transformation is still worth working on it ? (i.e. not cleaned or something else) 
This is for example something which is done in the RMS to handle canceled requests

Reply:
> Oh I see. Wouldn't it be easier in step 5. to have these agents checking that the Transformation is still worth working on it ? (i.e. not cleaned or something else)

No, and answer I wrote above: 
> Of course, one could make one final check in TransformationAgent or WorkflowTaskAgent but these 2 agents are already doing a lot of stuff, and are pretty heavy.